Parcare/Services/ParkingLotService.cs

```diff
@@ -41,9 +45,60 @@ public void ParkCar(string carNumber)
         }
         public bool IsCarParked(string carNumber)
         {
-            ParkingLotModel parkedCar = _parkedCarsList.Find(car => car.CarNumber == carNumber);
+            ParkingLotModel? parkedCar = _parkedCarsList.Find(car => car.CarNumber == carNumber);
             return (parkedCar != null);
         }
 
+        public bool IsPaymentReceived(string carNumber)
+        {
+            ParkingLotModel? parkedCar = _parkedCarsList.Find(car => car.CarNumber == carNumber);
+            if (parkedCar != null )
+                return parkedCar.PaymentReceived;
+            else
+            {
+                Console.WriteLine("There is no car with that number here.");
+                return false;
+            }
+        }
+
+        public void PayForParking(string carNumber)
+        {
+            ParkingLotModel? parkedCar = _parkedCarsList.Find(car => car.CarNumber == carNumber);
+            if(parkedCar != null)
+            {
+                
+                var duration = DateTime.Now - parkedCar.EntryTime;
+
+                Console.WriteLine(parkedCar.EntryTime);
+                if (duration.TotalHours <= 1)
+                {
+                    Console.WriteLine("Parking for an hour or less is free. Have a nice day!");
+                    parkedCar.PaymentReceived = true;
+                }
+                else
+                {
+                    // -1 for the first hour which is free
+                    var paymentAmount = (duration.TotalHours) * _price;
+                    var accountWithCarNumber = _bank.accounts.FirstOrDefault(account => account.CarNumber.Contains(carNumber));
+                    if(accountWithCarNumber != null)
```

Comment:
revert this if statement and if null just writeline and return, whatever is in if will be not nested anymore